src/train_and_predict.py

```diff
@@ -77,164 +84,210 @@
 def format_result(results, skip_keys={"Name"}):
     return '; '.join('{0}={1:g}'.format(k, v) for k, v in results.items() if k not in skip_keys)
 
-# Max F1 score
 def f1(p, r):
```

Comment:
Do you think it makes sense to pull all of the training/evaluating functions out into a separate script? And move `train_and_predict` and other SL prediction code to a subdirectory?

Reply:
Yes, but lets do this in another PR in which we make HANDL an installable python package. I'm having trouble figuring out how to do sensible relative imports from parent directories. I feel like this PR is getting too big and the purpose of this PR was to get the training code complete and working. Lets merge this change for now and I will refactor more in another PR.